### DISCUSSION TITLE
Server load explodes and crashes the server when trying to "review face groups"

### DISCUSSION THREAD

**Discussion by amo13** [matiasdelellis/facerecognition#707] [Q&A] [answered]
Unfortunately, it's been a while since I am unable to use the face recognition app. When I go to my personal settings and choose face recognition, I can see the recognized people. Then I click on one of them. Loading takes a bit of time because I have many many photos, but that's okay. When I click on "review face groups" to sort out false positives for that person, I can see a few thumbnail pictures appear but it's mostly the grey shapes staying that way:
![Screenshot from 2023-10-30 21-17-00](https://github.com/matiasdelellis/facerecognition/assets/6617661/a19f6cf6-a7e5-40db-a175-13d424f13cef)

If I take a look at `htop` on my server, I can see the load explode:
![Screenshot from 2023-10-30 21-13-14](https://github.com/matiasdelellis/facerecognition/assets/6617661/cf384f79-c42e-41e9-9289-941e3b35bd88)
and some time later:
![Screenshot from 2023-10-30 23-16-43](https://github.com/matiasdelellis/facerecognition/assets/6617661/a5133d53-9927-45cf-ad70-448974a0d6b3)

Even if I close the browser tab, my server stays unresponsive for a few hours. Before setting `CPUQuota=300%`, `MemoryHigh=50%` and `MemoryMax=66%` for the systemd service, even htop would freeze and I would not be able to ssh into the server for the few hours.

I suspected that some php bad configuration is the culprit but I couldn't find what to change. What's interesting to me is that with htop it looks like redis is the bottleneck. But I have no idea what is actually going on in the background when loading that page. 
I'd appreciate any help with dealing with that issue.

**Comment by matiasdelellis** [comment]
Hi  @amo13 
Sorry for delay.. I'm very busy.. 😞 
Please, try [Memories](https://apps.nextcloud.com/apps/memories). It should be much more efficient. 😉

**Reply by amo13** [reply]
Thanks for your response despite your lack of time, I appreciate it. 
I tried and failed because of a [Memories app issue](https://github.com/pulsejet/memories/issues/734). But even if it gets fixed, does Memories allow for easily sorting out false positives for a given person with a lot of photos?

**Reply by matiasdelellis** [reply, marked as answer]
About this [issue](https://github.com/pulsejet/memories/issues/734), (I should fix it when I can), there is a trick. You can assign the name directly. (Unfortunately without autocompletion), but if you assign the name the clusters are automatically merged.

> Memories allow for easily sorting out false positives for a given person with a lot of photos?

You can use the "Mark person in the preview" option and in any case fix it with the person sidebar

**Reply by amo13** [reply]
Ah, great that there is a workaround for the time being. Thank you very much for your help with this!

**Comment by 36grad** [comment]
Coming a bit late to the party, but a workaround for the OP's issue for people who use Nextcloud with apache2 is to set MaxClients to a suitable number in mpm_prefork.conf (probably there is a similar solution with nginx). 
Essentially the web UI will spawn a process for each of the pictures that need to be loaded. On my machine, each process would consume about ~100MB RAM, so with the default of 256 max client connections (effectively processes), that overloads my machine and leaves me with a lot of gray shapes.
Limiting the MaxClients to e.g. 40 will cause my user to create a web server RAM consumption of not more than 4GB which is for my family use case very effective. I have not noticed any negative side effects.

I guess somewhere in the javascript of the UI there should be a limit / queue so that not for each visible frame the data is fetched immediately from the web server, but only a limited number of items at a time.
